src/server/routers/project.ts

```diff
@@ -1,16 +1,155 @@
-import { Stage } from "@prisma/client";
+import { Role, Stage } from "@prisma/client";
 import { z } from "zod";
 
+import { nullable } from "@/lib/utils/general/nullable";
 import { stageCheck } from "@/lib/utils/permissions/stage-check";
 import { instanceParamsSchema } from "@/lib/validations/params";
+import { updatedProjectSchema } from "@/lib/validations/project-form";
 
 import {
   createTRPCRouter,
   protectedProcedure,
   stageAwareProcedure,
 } from "@/server/trpc";
+import { createManyFlags } from "@/server/utils/flag";
+import { updateProjectAllocation } from "@/server/utils/project-allocation";
 
 export const projectRouter = createTRPCRouter({
+  getEditFormDetails: protectedProcedure
```

Comment:
Naming of getEditFormDetails a bit unclear, returns the capacityUpperBound and preAllocatedStudentId as string.

Reply:
Sorted in last commit - removed procedure.